src/Volume.ts

```diff
@@ -157,146 +127,92 @@ export default class Volume {
+      const channel = new Channel(this.channelNames[i]);
       this.channels.push(channel);
       // TODO pass in channel constructor...
-      channel.dims = [this.x, this.y, this.z];
+      channel.dims = this.imageInfo.regionSize.toArray();
     }
 
-    this.physicalUnitSymbol = this.imageInfo.pixel_size_unit;
+    this.physicalUnitSymbol = this.imageInfo.spatialUnit;
     this.tickMarkPhysicalLength = 1;
-    this.setVoxelSize(this.pixel_size);
-
-    // make sure a transform is specified
-    if (!this.imageInfo.transform) {
-      this.imageInfo.transform = {
-        translation: [0, 0, 0],
-        rotation: [0, 0, 0],
-      };
-    }
-    if (!this.imageInfo.transform.translation) {
-      this.imageInfo.transform.translation = [0, 0, 0];
-    }
-    if (!this.imageInfo.transform.rotation) {
-      this.imageInfo.transform.rotation = [0, 0, 0];
-    }
+    this.setVoxelSize(this.imageInfo.pixelSize);
 
     this.volumeDataObservers = [];
   }
 
+  private validatePixelSize() {
+    this.imageInfo.pixelSize.x = this.imageInfo.pixelSize.x || 1.0;
+    this.imageInfo.pixelSize.y = this.imageInfo.pixelSize.y || 1.0;
+    this.imageInfo.pixelSize.z = this.imageInfo.pixelSize.z || 1.0;
+  }
+
   updateDimensions() {
-    this.x = this.imageInfo.tile_width;
-    this.y = this.imageInfo.tile_height;
-    this.z = this.imageInfo.tiles;
-    this.setVoxelSize(this.pixel_size);
-
-    /* eslint-disable @typescript-eslint/naming-convention */
-    const {
-      tile_width,
-      tile_height,
-      tiles,
-      vol_size_x = tile_width,
-      vol_size_y = tile_height,
-      vol_size_z = tiles,
-      offset_x = 0,
-      offset_y = 0,
-      offset_z = 0,
-    } = this.imageInfo;
-    /* eslint-enable @typescript-eslint/naming-convention */
-    this.contentSize = new Vector3(tile_width / vol_size_x, tile_height / vol_size_y, tiles / vol_size_z);
-    this.contentOffset = new Vector3(offset_x / vol_size_x, offset_y / vol_size_y, offset_z / vol_size_z);
+    const { pixelSize, volumeSize, regionSize, regionOffset } = this.imageInfo;
+
+    this.setVoxelSize(pixelSize);
+
+    this.normRegionSize = regionSize.clone().divide(volumeSize);
+    this.normRegionOffset = regionOffset.clone().divide(volumeSize);
   }
 
   // we calculate the physical size of the volume (voxels*pixel_size)
   // and then normalize to the max physical dimension
-  setVoxelSize(values: number[]): void {
-    // basic error check.  bail out if we get something bad.
-    if (!values.length || values.length < 3) {
-      return;
-    }
-
+  setVoxelSize(size: Vector3): void {
     // only set the data if it is > 0.  zero is not an allowed value.
-    if (values[0] > 0) {
-      this.pixel_size[0] = values[0];
-    }
-    if (values[1] > 0) {
-      this.pixel_size[1] = values[1];
-    }
-    if (values[2] > 0) {
-      this.pixel_size[2] = values[2];
-    }
+    // TODO this indicates that maybe pixel size should stick around?
```

Comment:
I suspect this comment will need more clarification

Reply:
This is referring to the fact that I deleted the `Volume` property that held onto pixel size, but this method is explicitly for modifying pixel size, so if we want `ImageInfo` to be a read-only structure it ought to be brought back out.

(I'm still not sure when we'd want to call this method from outside?)